grype/cpe/cpe.go

```diff
@@ -1,19 +1,22 @@
 package cpe
 
 import (
+	"github.com/anchore/grype/internal/log"
 	"github.com/anchore/syft/syft/pkg"
 )
 
 func NewSlice(cpeStrs ...string) ([]pkg.CPE, error) {
-	ret := make([]pkg.CPE, len(cpeStrs))
-	for idx, c := range cpeStrs {
+	var cpes []pkg.CPE
+	for _, c := range cpeStrs {
 		value, err := pkg.NewCPE(c)
 		if err != nil {
-			return nil, err
+			log.Warnf("unable to hydrate CPE for string %q, omitting from result CPE slice: %v", c, err)
```

Comment:
maybe "hydrate" -> "parse" ? or less verbose: "Excluding invalid CPE %q: %v"?

Reply:
@kzantow I like that! I'll create a PR for this